worker/upload.go

```diff
@@ -543,40 +539,26 @@ func (mgr *uploadManager) Upload(ctx context.Context, r io.Reader, contracts []a
 				case <-ctx.Done():
 				}
 				return
-			} else if up.packing && errors.Is(err, io.ErrUnexpectedEOF) {
-				mem.Release()
-
-				// uploadPacking is true, we return the partial slab without
-				// uploading.
-				partialSlab = data[:length]
+			} else if errors.Is(err, io.ErrUnexpectedEOF) && up.packing {
+				go mgr.addPartialSlab(ctx, mem, up.rs, data[:length], up.contractSet, respIndex, respChan)
+				respIndex++
 			} else {
-				// regular upload
-				go func(rs api.RedundancySettings, data []byte, length, slabIndex int) {
-					uploadSpeed, overdrivePct := upload.uploadSlab(ctx, rs, data, length, slabIndex, respChan, mgr.candidates(upload.allowed), mem, mgr.maxOverdrive, mgr.overdriveTimeout)
-
-					// track stats
-					mgr.statsSlabUploadSpeedBytesPerMS.Track(float64(uploadSpeed))
-					mgr.statsOverdrivePct.Track(overdrivePct)
-
-					// release memory
-					mem.Release()
-				}(up.rs, data, length, slabIndex)
+				go mgr.addSlab(ctx, mem, up.rs, upload, data, length, respIndex, respChan)
```

Comment:
Same issue with the `Release`. Also I don't think `addSlab` on `mgr` is a great name for what is happening. If anywhere I'd expect this to be a method on the `upload` as it was before, but without a `mem`, `up.rs`, `length`, `respIndex` or `respChan`. All of that could be stuff that the `upload` already knows about.
I get that you want to get some code out of the `if` `else` cases but I'm not sure more methods make the upload code any cleaner. I'd honestly keep this the way it is until we revisit this when implementing rhpv4 with more time to rethink the naming and structure.